src/app/static/src/app/store/modelOutput/mutations.ts

```diff
@@ -3,12 +3,34 @@ import {ModelOutputState} from "./modelOutput";
 import {PayloadWithType} from "../../types";
 
 export enum ModelOutputMutation {
-    TabSelected = "TabSelected"
+    TabSelected = "TabSelected",
+    Update = "Update"
 }
 
 export const mutations: MutationTree<ModelOutputState> = {
 
     [ModelOutputMutation.TabSelected](state: ModelOutputState, tab: string) {
         state.selectedTab = tab;
+    },
+
+    [ModelOutputMutation.Update](state: ModelOutputState, payload: any) {
+        const {selectedTab, selections} = state
+        // As the barchart does not update on render, any updates from the bar tab are true updates
+        if (selectedTab === 'bar'){
+            state.changes = true;
+        } else {
+            // As the bubble and map update on render, this stores their initial updates in state
+            // and then references new updates against those initially logged inorder to detect
+            // changes made by the user
+            const originKey = Object.keys(payload)[0]
+            const originTab = originKey + '_' + selectedTab
```

Comment:
If we keep this bit of code: Can you explain the significance of originKey please, and the need for the conditional on l.28? Not really sure what the need for it is. If you need to clone the object in some cases (?), would it be easier/safer just to do that in all cases, and cut down the complexity of this code? 

Isn't the payload the full dynamic form config here with hierarchical data? So unlikely to be a string or number? Should it be doing a deepClone? (I think we have a util method for that somewhere.)